shared/macros/10-bash.jinja

```diff
@@ -1040,6 +1042,88 @@ else
 fi
 {{%- endmacro -%}}
 
+{{#
+    Validate an authselect custom profile.
+    This sequence of commands is used in multiple places. This macro avoids duplicated code.
+
+    :param pam_file:    PAM config file.
+#}}
+{{%- macro bash_validate_authselect_custom_profile(pam_file) -%}}
+{{{ bash_check_authselect_integrity() }}}
+{{{ bash_ensure_authselect_custom_profile() }}}
+PAM_FILE_NAME=$(basename "{{{ pam_file }}}")
+PAM_FILE_PATH="/etc/authselect/$CURRENT_PROFILE/$PAM_FILE_NAME"
```

Comment:
Where is this Bash variable used? It does not seem to be used in this macro.
I also see this line and the previous one was present in some macros below, but you removed them as part of refactoring... is that expected?

Reply:
This macro is always used inside other macros and never alone. This bash variable, for example, is used after in the `bash_ensure_pam_module_line` macro.

For context, this chunk of code was repeated in multiple places. This was the reason I created the `bash_validate_authselect_custom_profile` macro as part of the refactoring. So, this is expected.

Reply:
I understand the reason why you created the new macro, but I think moving those two lines decreases readability.
If I understand it correctly:
On line 2150, the variable PAM_FILE_PATH gets set.
But if the condition on the line 2151 evaluates to true, it gets redefined within a macro on line 1055. There is no hint at that place that some redefinition of the variable might happen.
It seems that the variable PAM_FILE_PATH does not seem to be used within the macro bash_validate_authselect_custom_profile, it gets only changed there.
I can imagine this can be quite a debugging nightmare, as someone is trying to find out where this variable got redefined.

Reply:
Hum, good point of view. This indeed is not good if the reader is not used to this macros. On the Ansible equivalent the macro name is more suggestive: `ansible_ensure_pam_facts_and_authselect_profile`.

I will update the macro name in alignment to the equivalent in Ansible so the macro name is auto explanatory.  Probably the macro description can also be improved. I will send an additional commit soon.

Reply:
Hm... I see that some macros modify some Bash variables, e.g. macro bash_ensure_authselect_custom_profile modifies the variable CURRENT_PROFILE. This CURRENT_PROFILE is used later to define the PAM_FILE_PATH but already outside that macro... I think it should not be like that. It is creating dependencies among macros which are hidden.
Do you think it would be possible to rewrite it so that those variable assignments are more explicit and preferably they happen together?
I believe code deduplication should not decrease its readability.

Reply:
Done in the last commit. Could you take a look, please?

Reply:
Hm... the more I read it the more I think that maybe this shouldn't be deduplicated. I imagine macro as something discrete, doing one thing and not having side effects.
@matejak  @yuumasato  @ggbecker  what do you think about this?

Reply:
It is true that the macro behaves correctly. My concerns are really mainly about readability. Especially about readibility of the Jinja macros, not expanded macros. HOw about adding a Jinja comments there?
For example after line 1121:
{{# the following macro also modifies the variable PAM_FILE_PATH }}
Because this is exactly the information I would be missing when reading the Jinja file.

Reply:
I agree with Vojta that defining or using bash variables from other macros reduces overall readability. At the same time, I a see a slight  value in deduplicating the code.

The other case I see where this happens is with `CURRENT_PROFILE`, @marcusburghardt are you aware of other examples?

I think the documentation of the macros should mention the Bash variables they are defining that other macros can use, and the Bash variables they are using from other macros.

Reply:
@vojtapolasek thanks for clarifying your point. I believe a comment after the line 1121, as you suggested, is valid and good to have in this case. I will include it where the variable is indeed modified. 

@yuumasato , you can see that variable is already documented in the macro description by the last commit.

Reply:
@vojtapolasek , I amended the last commit including the comments. Could you take a look, please?

Reply:
Hello, thank you, this is sufficient for me.
Please could you do the same for the macro bash_ensure_authselect_custom_profile? It also sets the CURRENT_PROFILE variable and it is not clear. Please ammend the macro description and add a Jinja comment to the place where the macro is used.

Reply:
I amended the last commit.

Reply:
Looks good now. Thank you.